src/System.Devices.Gpio/System.Devices.Gpio/UnixI2CDevice.cs

```diff
@@ -0,0 +1,349 @@
+            I2C_M_RD = 0x0001
+        }
+
+        private unsafe struct i2c_msg
+        {
+            public ushort addr;
+            public I2cMessageFlags flags;
+            public ushort len;
+            public byte* buf;
+        };
+
+        /// <summary>Used in the <see cref="I2C_RDWR"/> <see cref="ioctl"/> call</summary>
+        private unsafe struct i2c_rdwr_ioctl_data
+        {
+            public i2c_msg* msgs;
+            public uint nmsgs;
+        };
+
+        private enum SmbusMessageFlags : byte
+        {
+            /// <summary>Write data to slave</summary>
+            I2C_SMBUS_WRITE = 0,
+            /// <summary>Read data from slave</summary>
+            I2C_SMBUS_READ = 1
+        }
+
+        private enum SmbusTransactionType : uint
+        {
+            I2C_SMBUS_BLOCK_DATA = 5
+        }
+
+        private unsafe struct i2c_smbus_ioctl_data
+        {
+            public SmbusMessageFlags read_write;
+            public SmbusTransactionType size;
+            public byte command;
+            public byte* data;
+        }
+
+        /// <summary>As specified in SMBus standard</summary>
+        private const int I2C_SMBUS_BLOCK_MAX = 32;
+
+        [StructLayout(LayoutKind.Explicit)]
+        private unsafe struct i2c_smbus_data
+        {
+            [FieldOffset(0)]
+            public byte u8;
+
+            [FieldOffset(0)]
+            public ushort word;
+
+            /// <summary>block[0] is used for length and one more for user-space compatibility</summary>
+            [FieldOffset(0)]
+            public fixed byte block[I2C_SMBUS_BLOCK_MAX + 2];
+        }
+
+        [DllImport(LibraryName, SetLastError = true)]
+        private static extern int ioctl(int fd, uint request, IntPtr argp);
+
+        [DllImport(LibraryName, SetLastError = true)]
+        private static extern int ioctl(int fd, uint request, ulong argp);
+
+        [DllImport(LibraryName, SetLastError = true)]
+        private static extern int read(int fd, IntPtr buf, int count);
+
+        [DllImport(LibraryName, SetLastError = true)]
+        private static extern int write(int fd, IntPtr buf, int count);
+
+        #endregion
+
+        private enum TrasnferKind
+        {
+            Read,
+            Write
+        }
+
+        private int _deviceFileDescriptor = -1;
+        private I2cFunctionalityFlags _functionalities;
+
+        public UnixI2cDevice(I2cConnectionSettings settings)
+            : base(settings)
+        {
+        }
+
+        public override void Dispose()
+        {
+            if (_deviceFileDescriptor >= 0)
+            {
+                close(_deviceFileDescriptor);
+                _deviceFileDescriptor = -1;
+            }
+        }
+
+        private unsafe void Initialize()
+        {
+            if (_deviceFileDescriptor >= 0)
+            {
+                return;
+            }
+
+            string devicePath = $"/dev/i2c-{_settings.BusId}";
```

Comment:
Can the "/dev/i2c-" be a public property setting?  In case a device has a different path